enshrine-esdt-safe/interactor/tests/enshrine-esdt-safe_interactor_tests.rs

```diff
@@ -0,0 +1,185 @@
+use enshrine_esdt_safe_interactor::ContractInteract;
+use interactor::constants::{TOKEN_ID, WHITELIST_TOKEN_ID};
+use interactor::interactor_config::Config;
+use multiversx_sc_snippets::imports::*;
+use proxies::*;
+use transaction::*;
+
+type OptionalTransferData<M> =
+    OptionalValue<MultiValue3<GasLimit, ManagedBuffer<M>, ManagedVec<M, ManagedBuffer<M>>>>;
+
+#[tokio::test]
+#[ignore]
+async fn test_deposit_paused() {
+    let mut interact = ContractInteract::new(Config::load_config()).await;
+    interact.deploy_token_handler().await;
+    interact.deploy(false).await;
+    interact
+        .deposit(
+            OptionalTransferData::None,
+            Some(ExpectError(4, "Cannot create transaction while paused")),
+        )
+        .await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn test_deposit_no_payment() {
+    let mut interact = ContractInteract::new(Config::load_config()).await;
+    let to = interact.bob_address.clone();
+    let from = interact.wallet_address.clone();
+    let to_contract = interact.state.current_address().clone();
+    let transfer_data = OptionalTransferData::None;
+
+    interact.deploy_setup().await;
+
+    interact
+        .interactor
+        .tx()
+        .from(from)
+        .to(to_contract)
+        .gas(30_000_000u64)
+        .typed(enshrine_esdt_safe_proxy::EnshrineEsdtSafeProxy)
+        .deposit(to, transfer_data)
+        .returns(ExpectError(4, "Nothing to transfer"))
+        .run()
+        .await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn test_deposit_too_many_payments() {
+    let mut interact = ContractInteract::new(Config::load_config()).await;
+    let to = interact.bob_address.clone();
+    let from = interact.wallet_address.clone();
+    let to_contract = interact.state.current_address().clone();
+    let transfer_data = OptionalTransferData::None;
+    let payments = ManagedVec::from(vec![
+        (
+            TokenIdentifier::from_esdt_bytes(TOKEN_ID),
```

Comment:
Can you make a variable to populate this vector? Or maybe a getter function to return the whole vector? It is too much code in the test